packages/xrpl/src/utils/getNFTokenID.ts

```diff
@@ -0,0 +1,85 @@
+import flatMap from 'lodash/flatMap'
+
+import {
+  CreatedNode,
+  isCreatedNode,
+  isModifiedNode,
+  ModifiedNode,
+  TransactionMetadata,
+} from '../models/transactions/metadata'
+
+interface NFToken {
+  NFToken: {
+    NFTokenID: string
+    URI: string
+  }
+}
+
+/**
+ * Gets the NFTokenID for an NFT recently minted with NFTokenMint.
+ *
+ * @param meta - Metadata from the response to submitting an NFTokenMint transaction.
+ * @returns The NFTokenID for the minted NFT.
+ * @throws if meta is not TransactionMetadata.
+ */
+export default function getNFTokenID(
+  meta: TransactionMetadata,
+): string | undefined {
+  /* eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- Provides a nicer error for js users */
+  if (meta.AffectedNodes === undefined) {
+    throw new TypeError(`Unable to parse the parameter given to getNFTokenID. 
+    'meta' must be the metadata from an NFTokenMint transaction. Received ${JSON.stringify(
+      meta,
+    )} instead.`)
+  }
+
+  /*
+   * When a mint results in splitting an existing page,
+   * it results in a created page and a modified node. Sometimes,
+   * the created node needs to be linked to a third page, resulting
+   * in modifying that third page's PreviousPageMin or NextPageMin
+   * field changing, but no NFTs within that page changing. In this
+   * case, there will be no previous NFTs and we need to skip.
+   * However, there will always be NFTs listed in the final fields,
+   * as rippled outputs all fields in final fields even if they were
+   * not changed. Thus why we add the additional condition to check
+   * if the PreviousFields contains NFTokens
+   */
+
+  const affectedNodes = meta.AffectedNodes.filter((node) => {
+    if (isCreatedNode(node)) {
+      return node.CreatedNode.LedgerEntryType === 'NFTokenPage'
+    }
+    if (isModifiedNode(node)) {
+      return (
+        node.ModifiedNode.LedgerEntryType === 'NFTokenPage' &&
+        Boolean(node.ModifiedNode.PreviousFields?.NFTokens)
+      )
+    }
+    return false
+  })
+  /* eslint-disable @typescript-eslint/consistent-type-assertions -- Necessary for parsing metadata */
```

Comment:
We should find a way to eliminate all of these disabled rules.

Reply:
These linting errors don't show up in the explorer and I am pretty sure we have `@typescript-eslint/no-unnecessary-condition` turned on.

Reply:
I believe currently the rule is set to "No Type Assertions", we could override it to do "as" type assertions. (I don't think type assertions are unreasonable)

Alternatively the thought could be that having the "No Type Assertions" rule has a more positive impact on the code quality which is worth the cost of occasionally adding the linting directive. (There are about 15 files in the `xrpl` package with this rule disabled with directives)

I don't have a strong preference either way.

Reply:
:( Not a fan of that rule.

Reply:
@ckniffen I made this PR to remove it :) - https://github.com/XRPLF/xrpl.js/pull/2220